src/main/java/com/pmm/budgetcalculator/expensing/service/ExpenseService.java

```diff
@@ -34,17 +42,17 @@ public void removeExpenseById(Long id) {
         } else throw new ExpenseNotFoundException(id);
     }
 
-    public Expense updateExpense(Long id, Expense expenseDetails) {
+    public ExpenseDto updateExpense(Long id, ExpenseDto expenseDetailsDto) {
         if (expenseRepository.existsById(id)) {
+            Expense expense = expenseMapper.dtoToEntity(expenseDetailsDto);
             Expense updateExpense = expenseRepository.getExpenseById(id);
-            updateExpense.setExpenseAmount(expenseDetails.getExpenseAmount());
-            updateExpense.setExpensePlace(expenseDetails.getExpensePlace());
-            updateExpense.setExpenseTypes(expenseDetails.getExpenseTypes());
-            updateExpense.setExpenseName(expenseDetails.getExpenseName());
-            updateExpense.setExpenseDescription(expenseDetails.getExpenseDescription());
-            updateExpense.setExpenseTime(expenseDetails.getExpenseTime());
-           return  expenseRepository.save(expenseDetails);
-
+            updateExpense.setExpenseAmount(expenseDetailsDto.getExpenseAmount());
+            updateExpense.setExpensePlace(expenseDetailsDto.getExpensePlace());
+            updateExpense.setExpenseName(expenseDetailsDto.getExpenseName());
+            updateExpense.setExpenseDescription(expenseDetailsDto.getExpenseDescription());
+            updateExpense.setExpenseTime(expenseDetailsDto.getExpenseTime());
+            ExpenseDto expenseDetails = expenseMapper.entityToDto(expense);
+            return expenseMapper.entityToDto(expense);
         } else throw new ExpenseNotFoundException(id);
```

Comment:
Although is not incorrect there is unwritten contract that java devs always add {} in ifs, elses, try, catches, loop etc.